examples/SelectiveHistorian/sqlhistorian/historian.py

```diff
@@ -0,0 +1,399 @@
+                    if not(topic_on_list): topics_blocked = topics_blocked+1
+
+                if not(self.filter_topics_enable) or topic_on_list:
+
+                    # look at the topics that are stored in the database
+                    # already to see if this topic has a value
+                    lowercase_name = topic.lower()
+                    topic_id = self.topic_id_map.get(lowercase_name, None)
+                    db_topic_name = self.topic_name_map.get(lowercase_name,
+                                                            None)
+                    _log.debug('topic is {}, db topic is {}'.format(
+                        topic, db_topic_name))
+                    if topic_id is None:
+                        _log.debug('Inserting topic: {}'.format(topic))
+                        # Insert topic name as is in db
+                        row = self.writer.insert_topic(topic)
+                        topic_id = row[0]
+                        # user lower case topic name when storing in map
+                        # for case insensitive comparison
+                        self.topic_id_map[lowercase_name] = topic_id
+                        self.topic_name_map[lowercase_name] = topic
+                        _log.debug('TopicId: {} => {}'.format(topic_id, topic))
+                    elif db_topic_name != topic:
+                        _log.debug('Updating topic: {}'.format(topic))
+                        self.writer.update_topic(topic, topic_id)
+                        self.topic_name_map[lowercase_name] = topic
+
+                    old_meta = self.topic_meta.get(topic_id, {})
+                    if set(old_meta.items()) != set(meta.items()):
+                        _log.debug(
+                            'Updating meta for topic: {} {}'.format(topic,
+                                                                    meta))
+                        self.writer.insert_meta(topic_id, meta)
+                        self.topic_meta[topic_id] = meta
+
+                    if self.writer.insert_data(ts, topic_id, value):
+                        # _log.debug('item was inserted')
+                        real_published.append(x)
+
+            if len(real_published) > 0:
+                if self.writer.commit():
+                    _log.debug('published {} data values'.format(
+                        len(to_publish_list)-topics_blocked)) #MODIFIED TO FILTER TOPICS
+                    self.report_all_handled()
+                else:
+                    msg = 'commit error. rolling back {} values.'
+                    _log.debug(msg.format(len(to_publish_list)))
+                    self.writer.rollback()
+            elif topics_blocked < len(to_publish_list): #MODIFIED TO FILTER TOPICS
+                _log.debug(
+                    'Unable to publish {}'.format(len(to_publish_list)-topics_blocked)) #MODIFIED TO FILTER TOPICS
+        except:
```

Comment:
- To catch any exception use the ```except Exception```, instead of ```except BaseException``` or just ```except```.
- Indicate the most specific exception type in ```except``` block.

For example:
```python
# Bad
try:
    mapping[key]
except Exception:
    ...

# Better
try:
    mapping[key]
except KeyError:
    ...
```